File: DONGUK/프로그래머스_WEEK_12/숫자블록.py
import math

def isPrime(n):
    for i in range(2, int(math.sqrt(n) + 1)):
        if n % i == 0:
            return False
    return True

def solution(begin, end):
    answer = []
        
    for num in range(begin, end+1):
        q = 0
        
        if num == 1:
            answer.append(0)
            continue
        
        if isPrime(num):
            answer.append(1)
            continue
            
        else:
            for i in range(2, int(num ** 0.5)+1):
                if num % i == 0:
                    q = num // i
                    
                    if q > 10000000:
                        q = 1
                        continue
                    
                    else:
                        break
                         
            answer.append(q)
            
    return answer

File: JUNHEE/프로그래머스_week12/숫자블록.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

long long is_prime(long long num){
    for(long long i = 2; i*i <= num; i++){
        if(num % i == 0 && (num / i) <= 10000000) return num / i;
    }
    return 1;
}
vector<int> solution(long long begin, long long end) {
    vector<int> answer(end - begin + 1, 0);
    for(int i = begin, j = 0; i <= end; i++, j++){
        answer[j] = is_prime(i);
        
    }
    if(begin == 1) answer[0] = 0;
    return answer;
}
